tests/api/runtimes/base.py

```diff
@@ -0,0 +1,207 @@
+            name = env_variable["name"]
+            if name in expected_variables:
+                if expected_variables[name]:
+                    assert expected_variables[name] == env_variable["value"]
+                expected_variables.pop(name)
+
+        # Make sure all variables were accounted for
+        assert len(expected_variables) == 0
+
+    def _assert_v3io_mount_configured(self, v3io_user, v3io_access_key):
+        args, _ = get_k8s().v1api.create_namespaced_pod.call_args
+        pod_spec = args[1].spec
+        container_spec = pod_spec.containers[0]
+
+        pod_env = container_spec.env
+        self._assert_env_variables_set_correctly_in_pod_env(
+            pod_env,
+            {
+                "V3IO_API": None,
+                "V3IO_USERNAME": v3io_user,
+                "V3IO_ACCESS_KEY": v3io_access_key,
+            },
+        )
+
+        expected_volume = {
+            "flexVolume": {
+                "driver": "v3io/fuse",
+                "options": {"accessKey": v3io_access_key},
+            },
+            "name": "v3io",
+        }
+        assert pod_spec.volumes[0] == expected_volume
+
+        expected_volume_mounts = [
+            {"mountPath": "/v3io", "name": "v3io", "subPath": ""},
+            {"mountPath": "/User", "name": "v3io", "subPath": f"users/{v3io_user}"},
+        ]
+        assert container_spec.volume_mounts == expected_volume_mounts
+
+    def _assert_pod_create_called(
+        self,
+        expected_params={},
+        expected_inputs={},
+        expected_hyper_params={},
+        expected_secrets={},
+        expected_limits={},
+    ):
+        create_pod_mock = get_k8s().v1api.create_namespaced_pod
+        create_pod_mock.assert_called_once()
+        args, _ = create_pod_mock.call_args
+        # assert args[0] == self.namespace
+        pod_spec = args[1]
+        self._assert_labels_exist_in_pod_creation(pod_spec.metadata.labels)
+
+        container_spec = pod_spec.spec.containers[0]
+        if expected_limits:
+            assert container_spec.resources["limits"] == expected_limits
+
+        pod_env = container_spec.env
+
+        self._assert_env_variables_set_correctly_in_pod_env(
+            pod_env, {"MLRUN_NAMESPACE": self.namespace}
+        )
+        for env_variable in pod_env:
+            if env_variable["name"] == "MLRUN_EXEC_CONFIG":
+                function_config = json.loads(env_variable["value"])
+                self._assert_function_config_as_expected(
+                    function_config,
+                    expected_params,
+                    expected_inputs,
+                    expected_hyper_params,
+                    expected_secrets,
+                )
+
+        assert pod_spec.spec.containers[0].image == self.image_name
+
+        # print(f"POD spec:\n {args[1]}")
+
+    # Needed for tracking pod status/logs
+    @staticmethod
+    def _mock_get_pod():
+        get_k8s().v1api.read_namespaced_pod = unittest.mock.Mock()
```

Comment:
Obviously at some point we would want to have a real response mock here (and test the different behaviors that happen beacuse of the different responses)

Reply:
I'm removing this one. Thought I'd need it, but I don't.